pandas/tests/indexes/multi/test_get_set.py

```diff
@@ -329,3 +329,21 @@ def test_set_levels_with_iterable():
         [expected_sizes, colors], names=["size", "color"]
     )
     tm.assert_index_equal(result, expected)
+
+
+def test_at_indexing_fails_multiindex():
+    # GH9259
+    cols = [("a_col", chr(i + 65)) for i in range(2)]
+    idx = [("a_row", chr(i + 65)) for i in range(2)]
+    df = pd.DataFrame(
+        np.linspace(1, 4, 4).reshape(2, 2),
+        index=pd.MultiIndex.from_tuples(idx),
+        columns=pd.MultiIndex.from_tuples(cols),
+    )
+    ser = df[("a_col", "A")]
+
+    with pytest.raises(KeyError, match=r".+? indexing with .+? failing for MultiIndex"):
+        df.at["a_row", "A"]
+
+    with pytest.raises(TypeError, match=r".+? got multiple values for argument .+?"):
+        ser.at["a_row", "A"]
```

Comment:
this looks like a bug.  `ser.at["a_row", "A"]` should return 1.0, same as `ser.loc["a_row", "A"]`

Reply:
I tested this on my PR #32520, it returns 1.0.

Reply:
I can remove the series from the test. @rhshadrach is correct that the original context was DataFrames, and improving the error message should clarify why the KeyError is being raised.